diff --git a/examples/cell_server.go b/examples/cell_server.go
index 1b0c981..cd1c51b 100644
--- a/examples/cell_server.go
+++ b/examples/cell_server.go
@@ -87,16 +87,20 @@ func Poller(in <-chan *Request, cell *cell.CellAdvisor, thread_number int) {
 				sendResult(done, r.result, []byte{})
 			case "screen":
 				go func() {
-					screenCache.mu.Lock()
-					defer screenCache.mu.Unlock()
 					if time.Now().Sub(screenCache.last).Seconds() > 1 {
-						screenCache.last = time.Now()
-						screenCache.cache, err = cell.GetScreen()
-						if err != nil {
-							log.Println(err.Error())
+						screenCache.mu.Lock()
+						defer screenCache.mu.Unlock()
+						if time.Now().Sub(screenCache.last).Seconds() > 1 {
+							screenCache.last = time.Now()
+							screenCache.cache, err = cell.GetScreen()
+							if err != nil {
+								log.Println(err.Error())
+							}
 						}
 					}
 
+					screenCache.mu.RLock()
+					defer screenCache.mu.RUnlock()
 					sendResult(done, r.result, screenCache.cache)
 				}()
 			case "heartbeat":
